src/main/java/fruitshop/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package fruitshop.service.impl;
+
+import fruitshop.model.FruitTransaction;
+import fruitshop.service.ShopService;
+import fruitshop.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction fruit : transactions) {
+            operationStrategy.getOperation(fruit);
```

Comment:
It seems you're retrieving an operation handler using `operationStrategy.getOperation(fruit)` but not executing any operation with it. According to the Strategy pattern, you should execute the operation that this handler is responsible for. You should invoke the method that performs the operation on the handler returned by the strategy.